_posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md

```diff
@@ -9,18 +9,27 @@ categories:
 tags: []
 ---
 
-Recently, I've worked on a [project](https://github.com/agrc/wfrc-wab-widgets/) building custom Web AppBuilder (WAB) widgets for a client. This has been an interesting digression from our typical development projects. Working within the confines of WAB has proved to be both liberating and frustrating. Being able to focus on a single widget and not have to worry about an entire application has been great. I've loved letting Esri take care of all of the basic application stuff such as loading the application framework, web map and widgets. However, not having control over the entire system has created some headaches. In this post, I'd like to cover some solutions that have helped me.
+Recently, I have been working on a [project](https://github.com/agrc/wfrc-wab-widgets/) building custom Web AppBuilder (WAB) widgets for a client. This has been an interesting digression from our typical development projects, and working within the confines of WAB has proved to be both liberating and frustrating. In this post, I would like to share a few solutions that have helped me make the most of working with WAB.
+
+## The Pros and Cons of Working with WAB Widgets
+But before we get to those tips, let's talk about some of the general pros and cons of working with WAB that I have discovered through this project. First, being able to focus on a single widget and not having to worry about an entire application has been great. I have loved letting Esri take care of all of the basic application stuff, such as loading the application framework, web map, and widgets. However, not having control over the entire system has created some headaches. That's where the WAB solutions came in.
+
+## Tips for Building WAB Widgets
```

Comment:
Again, I'm thinking subheadings with the specific tips might make the article more easily digestible at a glance? What are your thoughts on this @stdavis ?